### PR TITLE
♻️ Move logic from .travis.yml to bash scripts.

### LINKED ISSUE #2225: Move environment determination from .travis.yml to bash scripts
To avoid merge conflicts for `.travis.yml` it might be worthwhile to move the environment checks (`[ $TRAVIS_BRANCH == "production-amp-dev" ]; then unbuffer gulp buildPrepare; fi` and others) to a bash script.

### PR DESCRIPTION
**Pull request by matthiasrohmer**
Fixes #2225, foundation for #2163.

### CODE REVIEW

File: travis_scripts/deploy.sh [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+  rm -rf $HOME/google-cloud-sdk
+  export CLOUDSDK_CORE_DISABLE_PROMPTS=1
+  curl https://sdk.cloud.google.com | bash > /dev/null
+  source /home/travis/google-cloud-sdk/path.bash.inc
+  gcloud version --quiet
+  gcloud components update --quiet
+  gcloud components install beta --quiet
+  gcloud auth activate-service-account --key-file=.prod.key.json
+  gcloud config set project amp-dev-230314
```
Comment (sebastianbenz) [inline travis_scripts/deploy.sh]: Sidenote: is the gcloud setup cached in travis?

File: travis_scripts/deploy.sh [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+fi
+
+echo "Finalizing build ..."
+unbuffer gulp buildFinalize
+
+echo "Starting deployment ..."
+if [ "${TRAVIS_BRANCH}" = "production" ]; then
```
Comment (sebastianbenz) [inline travis_scripts/deploy.sh]: Shouldn't this be amp-dev-production?

File: travis_scripts/preparation.sh [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the license.
+
+set -e
+
+if [ "${TRAVIS_BRANCH}" = "production" ]; then
+  echo "Preparing build ..."
+  unbuffer gulp buildPrepare
+elif [ "${TRAVIS_BRANCH}" = "future" ] && [ $TRAVIS_PULL_REQUEST == "false" ]; then
```
Comment (sebastianbenz) [inline travis_scripts/preparation.sh]: Does it make sense to merge these two branches?

File: travis_scripts/before_install.sh [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the license.
+
+set -e
+
+if [ "${TRAVIS_BRANCH}" = "production" ]; then
```
Comment (sebastianbenz) [inline travis_scripts/before_install.sh]: It production correct here?

**Comment by matthiasrohmer** [comment]
@tharders, would be great if you could have another look 🙂

**Pull request merged**